cmake/modules/BuildDPDK.cmake

```diff
@@ -71,11 +71,13 @@ function(do_build_dpdk dpdk_dir)
       "\"${target}\" not listed in ${supported_targets}")
   endif()
 
+  set(GCC9_FLAGS "-Wno-stringop-truncation -Wno-address-of-packed-member -fPIC")
```

Comment:
"stringop-truncation" was added in GCC-8. while "address-of-packed-member" was added in GCC-9.

so the variable name is kinda misleading. also, have you tried to build the tree with GCC-7 or the lowest Clang version which compiles nautilus without this patch ? if not, shall we enable these CFLAGS only if the compiler supports them?

please see `BuildSPDK.cmake` in master branch for a reference.

also, you need to note down the reason why this change is not cherry-picked from master in the commit message.

could you paste the compiler warnings/errors you are addressing in the comment as a reference?

Reply:
> "stringop-truncation" was added in GCC-8. while "address-of-packed-member" was added in GCC-9.
> 
> so the variable name is kinda misleading. 

will fix the name.

> also, have you tried to build the tree with GCC-7 or the lowest Clang version which compiles nautilus without this patch ? if not, shall we enable these CFLAGS only if the compiler supports them?
> 
Didn't try the tree, just tried a test compilation. for g++7.3 (the default version on fedora27) the ``-Wno-unknown-warning-option`` is set by default, which makes it future proof for disabling warning flags. For clang++5.02 (the default version on fedora27) you have to set that explicitly, so, I will add that in the cmake file.

> please see `BuildSPDK.cmake` in master branch for a reference.
> 
> also, you need to note down the reason why this change is not cherry-picked from master in the commit message.
> 

I dont understand - the change does not exist in master. 

> could you prefix the title of your commit message with the subcomponent your are changing ? see https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#3-describe-your-changes. in this case, it's "cmake/BuildDPDK:"
> 
ok
> and the title of commit message should summarize the change made in the commit. see https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#3-describe-your-changes . in this change, it reads like a bug report.
> 
ok
> could you paste the compiler warnings/errors you are addressing in the comment as a reference?

ok

Reply:
> I dont understand - the change does not exist in master.

normally we cherry-pick fixes from master. if a change does not exist in master, we always put the reason in the commit message. like

```
Conflicts: cmake/modules/BuildDPDK.cmake
  this change is not cherry-picked from master, because master has picked up an updated SPDK which is in turn using a DPDK containing fixes like a385972c, 3370975b,  74f27710 and e5e193ac and a36b40dd. these changes either disables the "address-of-packed-member" warnings or addresses "error=stringop-truncation" warnings.
```

please check the output of `git log` and search for "not cherry-pick" for more examples.

Reply:
@tchaikov, thanks, going to take that comment :-)